docs: add warning in CSS in JS guide about CSS injection order (#3539)
aws-amplify/amplify-ui: docs/src/pages/[platform]/guides/css-in-js/web.mdx

> import { Alert, Link, Tabs, TabItem } from '@aws-amplify/ui-react';

Reviewer: nit: Link import is not used

Author: Fixed.

One of the goals of Amplify is to be a good citizen of the environment it exists in. As such Amplify works hard to not interfere with other tools that are being used.

There are many CSS In JS libraries out there that give a number of benefits. Amplify does not use a CSS In JS library to implement its theming system but operates well with the major CSS In JS libraries out there. Below are a few of the libraries with descriptions of how they work with Amplify's theming system.

<Alert variation="warning" role="none">

>   If your CSS library depends on injecting style tags into the DOM, make sure they are being injected <b>after</b> Amplify UI's style tag. In other words, ensure the style tag for Amplify UI is listed above your custom styles in the head tag. If Amplify UI's styles are injected after, then your custom styles will likely be unset. See <Link href="https://github.com/aws-amplify/amplify-ui/issues/3533">here</Link> for more info.

Reviewer 2: We don't have precendent for linking to specific issue from the docs. What do you think of this?

```suggestion
  If your CSS library depends on injecting style tags into the DOM, make sure they are being injected <b>after</b> Amplify UI's style tag. In other words, ensure the style tag for Amplify UI is listed above your custom styles in the `<head>` tag. If Amplify UI's styles are injected after custom styles, then your custom styles may be unset.
```

Author: Understood. This looks good to me!
